gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/ClassUniquenessLockTask.java

```diff
@@ -0,0 +1,155 @@
+
+                    if (problemJars.isEmpty()) {
+                        return Optional.empty();
+                    }
+
+                    StringBuilder stringBuilder = new StringBuilder();
+                    // TODO(dfox): ensure we iterate through problemJars in a stable order
+                    for (Set<ModuleVersionIdentifier> clashingJars : problemJars) {
+                        stringBuilder
+                                .append(clashingJars.stream()
+                                        .map(mvi -> mvi.getGroup() + ":" + mvi.getName())
+                                        .sorted()
+                                        .collect(Collectors.joining(", ", "[", "]")))
+                                .append('\n');
+
+                        analyzer.getDifferingSharedClassesInProblemJars(clashingJars).stream()
+                                .sorted()
+                                .forEach(className -> {
+                                    stringBuilder.append("  - ");
+                                    stringBuilder.append(className);
+                                    stringBuilder.append('\n');
+                                });
+                    }
+                    return Optional.of(stringBuilder.toString());
+                }));
+
+        boolean conflictsFound = resultsByConfiguration.values().stream().anyMatch(Optional::isPresent);
+        if (!conflictsFound) {
+            // this is desirable because if means if people apply the plugin to lots of projects which are already
+            // compliant, they don't get loads of noisy lockfiles created.
+            ensureLockfileDoesNotExist();
+        } else {
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.append("# Run ./gradlew checkClassUniquenessLock --write-locks to update this file\n\n");
+            // TODO(dfox): make configuration order stable!
+            resultsByConfiguration.forEach((configuration, contents) -> {
+                if (contents.isPresent()) {
+                    stringBuilder.append("## ").append(configuration).append("\n");
+                    stringBuilder.append(contents.get());
+                }
+            });
+            ensureLockfileContains(stringBuilder.toString());
+        }
+    }
+
+    private void ensureLockfileContains(String expected) {
+        if (getProject().getGradle().getStartParameter().isWriteDependencyLocks()) {
+            GFileUtils.writeFile(expected, lockFile);
+            getLogger().lifecycle("Updated {}", getProject().getRootDir().toPath().relativize(lockFile.toPath()));
+        } else {
+            String onDisk = GFileUtils.readFile(lockFile);
+            if (!onDisk.equals(expected)) {
+                throw new GradleException(lockFile
+                        + " is out of date, please run `./gradlew "
+                        + "checkClassUniquenessLock --write-locks` to update this file");
```

Comment:
I think we may want to provide a different message if new duplicates appear vs when problems are fixed, or mention here that it's dangerous to have duplicate classes on the classpath and explain why it's worth spending time to fix.
Otherwise I think folks will see "run X to update locks" and they'll `./X`.